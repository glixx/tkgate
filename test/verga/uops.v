module top;
  reg a,b;
  reg [31:0] x;
  reg [31:0] r;

 
  initial
    begin
      x = ~0;
      a = x;
      b = ~x;
      #1;
      $display("a=%h  b=%h",a,b);
      if (a) $display("a is true.");
      if (~a) $display("~a is true.");
      if (!a) $display("!a is true.");
      if (b) $display("b is true.");
      if (~b) $display("~b is true.");
      if (!b) $display("!b is true.");

      r = 32'hz;
      $display("r='h%h   r='o%o",r,r);
    end

endmodule

