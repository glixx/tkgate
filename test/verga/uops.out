a=1  b=0
a is true.
~b is true.
!b is true.
r='hz   r='oz
